tests/foreman/cli/test_template.py

```diff
@@ -74,6 +78,75 @@ def test_update_template_1(self):
         self.assertEqual(len(result.stderr), 0)
         self.assertEqual(updated_name, result.stdout['name'])
 
+    def test_create_template_with_location(self):
+        """@Test: Check if Template with Location can be created
+
+        @Feature: Template - Create
+
+        @Assert: Template is created and new Location has been assigned
+
+        """
+
+        name = gen_string("alpha", 10)
+
+        try:
+            new_loc = make_location()
+            new_template = make_template(
+                {
+                    'name': name,
+                    'location-ids': new_loc["id"],
+                }
+            )
+        except CLIFactoryError as e:
+            self.fail(e)
+
+        self.assertIn(new_loc['name'], new_template['locations'])
+
+    def test_create_template_locked(self):
+        """@Test: Check that locked Template cannot be created
+
+        @Feature: Template - Create
+
+        @Assert: It is not allowed to create locked Template
+
+        """
+
+        name = gen_string("alpha", 10)
+
+        with self.assertRaises(CLIFactoryError) as ex:
+            make_template(
+                {
+                    'name': name,
+                    'locked': 'true',
+                }
+            )
+        self.assertIn("This template is locked. Please clone it",
```

Comment:
There's only one issue with this assertion: if we were to change the locale of the system, then this message could potentially be localized and you would not get a message in english... since we currently don't test the CLI under different locales, this is not a blocker though, but something to consider.